src/Cas/Factories/ProcessingChainFactory.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the simplesamlphp-module-casserver.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace SimpleSAML\Module\casserver\Cas\Factories;
+
+use SimpleSAML\Auth\ProcessingChain;
+use SimpleSAML\Configuration;
+
+class ProcessingChainFactory
+{
+    /** @var Configuration */
+    private readonly Configuration $casconfig;
+
+    public function __construct(
+        Configuration $casconfig,
+    ) {
+        $this->casconfig = $casconfig;
+    }
+
+    /**
+     * @codeCoverageIgnore
+     * @throws \Exception
+     */
+    public function build(array $state): ProcessingChain
+    {
+        $idpMetadata = [
```

Comment:
I'm curious about the over lap with how this method is defing sp and idp metadata, and the overlap with `AttributeExtractor. runAuthProcs` where those are also defined. Do they have different purposes?

Reply:
> Do they have different purposes?

No. 

The reason why i am constructed them twice is because i only want to pass one parameter in the `build` function. I first push the values in the `state` and then i reconstruct them out of the `state`. Since there is no loop in the process and the code is really simple i thought it would be better to keep the function signature simpler in the expense of having 7 lines of duplicate code.

Reply:
but can't this just be `return new ProcessingChaing($state['Source'], $state['Destination'], 'casserver');` ?